bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/ephemeris/TcsEphemerisExport.scala

```diff
@@ -166,44 +168,42 @@ class TcsEphemerisExport(dir: File, night: Night, site: Site, odb: IDBDatabaseSe
   val start = new Date(night.getStartTime)
   val end   = new Date(night.getEndTime)
 
-  val lookupNonSiderealObservations: TryExport[HorizonsDesignation ==>> Set[String]] =
+  val lookupNonSiderealObservations: TryExport[Set[HorizonsDesignation]] =
     TryExport.fromTryCatch(OdbError) {
-      val ns = NonSiderealObservationFunctor.query(odb, user)
-      (==>>.empty[HorizonsDesignation, Set[String]]/:ns) { (m, n) =>
-        m.updateAppend(n.hid, Set(n.targetName))
-      }
+      NonSiderealObservationFunctor.query(odb, user).map(_.hid).toSet
     }
 
   val export: TryExport[HorizonsDesignation ==>> (ExportError \/ File)] =
     lookupNonSiderealObservations >>= exportAll
 
-  private def exportAll(nos: HorizonsDesignation ==>> Set[String]): TryExport[HorizonsDesignation ==>> (ExportError \/ File)] =
-    EitherT(nos.mapOptionWithKey { (hid, names) =>
-      // There can be multiple names for the same horizons id.  We'll just pick
-      // one at random for now ...
-      names.headOption.map { exportOne(dir, hid, _) }
-    }.traverse(_.run).map(_.right[ExportError]))
+  private def exportAll(nos: Set[HorizonsDesignation]): TryExport[HorizonsDesignation ==>> (ExportError \/ File)] = {
+    // HorizonsDesignation ==>> TryExport[File]
+    val exportMap = ==>>.fromList(nos.toList.map(hid => (hid, exportOne(dir, hid))))
+
+    // IO[HorizonsDesignation ==>> (ExportError \/ File)]
+    val exportOp  = exportMap.traverse(_.run)
 
-  def exportOne(dir: File, hid: HorizonsDesignation, name: String): TryExport[File] =
+    // TryExport[HorizonsDesignation ==> (ExportError \/ File)]
+    EitherT(exportOp.map(_.right[ExportError]))
+  }
+
+  def exportOne(dir: File, hid: HorizonsDesignation): TryExport[File] =
     for {
       em <- lookupEphemeris(hid)
-      f  <- writeEphemeris(hid, name, em)
+      f  <- writeEphemeris(hid, em)
     } yield f
 
   def lookupEphemeris(hid: HorizonsDesignation): TryExport[EphemerisMap] =
     HorizonsService2.lookupEphemerisE[EphemerisElement](hid, site, start, end, ElementCount) { (ee: EphemerisEntry) =>
       ee.coords.map((_, ee.getRATrack, ee.getDecTrack))
     }.leftMap(e => HorizonsError(hid, e): ExportError)
 
-  def writeEphemeris(hid: HorizonsDesignation, name: String, em: EphemerisMap): TryExport[File] = {
-    // TODO: taking a random string and assuming it is suitable for a filename
-    // TODO: seems like asking for trouble.  any suggestions?  URL encode?
-    // TODO: still not sure what to do about file names anyway.
-    val fileName = s"${name}_${hid.toString}.eph".replaceAll("/", "-")
-
+  def writeEphemeris(hid: HorizonsDesignation, em: EphemerisMap): TryExport[File] = {
+    // See TcsConfig.ephemerisFile
+    val fileName = URLEncoder.encode(s"${hid.toString}.eph", UTF_8.name)
```

Comment:
The `HorizonsDesignation` query can contain semicolons and spaces as well (at least .. there could be more) which will probably screw up the TCC, so we may need to do some additional encoding.